Pointers/UniquePtr.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+#include <iostream>
+
+using std::cout;
+using std::endl;
+
+template<typename T>
+class UniquePtr {
+public:
+	UniquePtr(T* ptr);
+	~UniquePtr();
+	// Don't allow copy
+	UniquePtr(const UniquePtr<T>& other) = delete;
+	UniquePtr<T>& operator=(const UniquePtr<T>& other) = delete;
+	// Allow move
+	UniquePtr(UniquePtr<T>&& other) noexcept;
+	UniquePtr<T>& operator=(UniquePtr<T>&& other) noexcept;
+
+	T& operator*() const;
+	T* operator->() const;
+private:
+	T* m_ptr;
+};
+
+template<typename T>
+UniquePtr<T>::UniquePtr(T* ptr) : m_ptr(ptr) {
+	cout << "UniquePtr ctor" << endl;
+}
+
+template<typename T>
+UniquePtr<T>::~UniquePtr() {
+	cout << "UniquePtr dtor" << endl;
+	if (m_ptr) {
+		cout << "delete UniquePtr m_ptr" << endl;
+		delete m_ptr;
+	}
+}
+
+template<typename T>
+UniquePtr<T>::UniquePtr(UniquePtr<T>&& other) noexcept {
+	// Check if other is the same object instance. If yes, don't do anything
+	if (this == &other) {
```

Comment:
This is only relevant for operator=, the ctor creates a new object so there is no way you for this condition to be true